modules/pricer/src/main/java/com/opengamma/strata/pricer/impl/rate/swap/CashFlowEquivalentCalculator.java

```diff
@@ -387,25 +387,43 @@ public static ResolvedSwapLeg cashFlowEquivalentOnLeg(
   }
 
   /**
-   * Generate a new list with the dates sorted and the amounts of elements with same payment date compressed.
+   * Extract the payments from the {@link NotionalExchange} in the SwapLeg.
+   * Generate a list with the dates sorted and the amounts of elements with same payment date compressed.
+   * <p>
+   * The original SwapLeg is unchanged.
+   * 
+   * @param input  the starting swap leg
+   * @return the normalized list
+   */
+  public static List<Payment> normalize(ResolvedSwapLeg input) {
+    List<Payment> cfe2 = new ArrayList<>();
+    for (SwapPaymentEvent cf : input.getPaymentEvents()) {
+      ArgChecker.isTrue(cf instanceof NotionalExchange, "the swap leg must consist of NotionalExchange");
+      cfe2.add(((NotionalExchange) cf).getPayment());
+    }
+    return normalize(cfe2);
+  }
+
+  /**
+   * Generate a new payment list with the dates sorted and the amounts of elements with same payment date compressed.
    * <p>
    * The original list is unchanged.
    * 
    * @param input  the starting list
    * @return the normalized list
    */
-  public static List<NotionalExchange> normalize(List<NotionalExchange> input) {
```

Comment:
as  this is a simple variable change and the algorithm is already in place and working, ok to let sleeping code lie...